src/pages/Details.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+export default function Details() {
+  const currencies = useSelector((state) => state.currency.allData);
+  const { currency } = useParams();
+  const [currencyValues, setCurrencyValues] = React.useState({});
+
+  function getKeyByValue(object, value) {
+    return Object.keys(object).find((key) => object[key] === value);
+  }
+
+  const key = getKeyByValue(currencies, currency);
+  const keyUpper = key && key.toUpperCase();
+
+  const date = new Date().toISOString().slice(0, 10);
+  React.useEffect(() => {
+    fetch(`https://cdn.jsdelivr.net/gh/fawazahmed0/currency-api@1/${date}/currencies/usd.json`)
+      .then((response) => response.json())
+      .then((data) => {
+        setCurrencyValues(data.usd);
+      });
+  }, [date]);
+
```

Comment:
**OPTIONAL** - Please note that in order to keep up with global JavaScript practices it is advisable to use ES6 in your project, hence you should use arrow functions instead. Please fix this 🙏

Reply:
It must have slipped out of my brain, with me focusing on the functionality itself. 
Thank you very much for your comment, Sir,